R/Model-validity.R

```diff
@@ -58,31 +58,53 @@ validate_logistic_log_normal <- function(object) {
     object@cov,
     mode = "numeric", nrows = 2, ncols = 2, any.missing = FALSE
   )
+  is_prec_2x2 <- test_matrix(
+    object@prec,
+    mode = "numeric", nrows = 2, ncols = 2, any.missing = FALSE
+  )
+  o$check(is_cov_2x2, "cov must be 2x2 matrix without any missing values")
+  o$check(is_prec_2x2, "prec must be 2x2 matrix without any missing values")
   if (is_cov_2x2) {
     o$check(
       h_is_positive_definite(object@cov),
       "cov must be positive-definite matrix"
     )
-  } else {
-    o$check(is_cov_2x2, "cov must be 2x2 matrix without any missing values")
+    if (is_prec_2x2) {
+      o$check(
+        all.equal(object@cov %*% object@prec, diag(1, 2), check.attributes = FALSE) == TRUE,
+        "prec must be inverse of cov"
+      )
+    }
   }
   o$check(
-    test_number(object@refDose, lower = 0 + .Machine$double.xmin),
-    "refDose must be positive scalar"
+    test_number(object@ref_dose, na.ok = TRUE, lower = 0 + .Machine$double.xmin),
+    "ref_dose must be positive scalar or NA_real_"
```

Comment:
hm I don't think we should allow `NA`. since that will give error in the JAGS then and we want to catch it before.

Reply:
then I would choose something like `ref_dose = 0`